File: src/main/java/com/koddy/server/coffeechat/infrastructure/link/MeetingLinkProcessor.java
package com.koddy.server.coffeechat.infrastructure.link;

import com.koddy.server.coffeechat.domain.model.link.MeetingLinkRequest;
import com.koddy.server.coffeechat.domain.model.link.MeetingLinkResponse;

public interface MeetingLinkProcessor {
    MeetingLinkResponse create(final String oAuthAccessToken, final MeetingLinkRequest meetingLinkRequest);

    void delete(final String meetingId);

    String OAUTH_CONTENT_TYPE = "application/json";

    String BEARER_TOKEN_TYPE = "Bearer";
}

File: src/main/kotlin/com/koddy/server/coffeechat/infrastructure/link/MeetingLinkProcessor.kt
package com.koddy.server.coffeechat.infrastructure.link

import com.koddy.server.coffeechat.domain.model.link.MeetingLinkRequest
import com.koddy.server.coffeechat.domain.model.link.MeetingLinkResponse

interface MeetingLinkProcessor {
    fun create(
        oAuthAccessToken: String,
        meetingLinkRequest: MeetingLinkRequest,
    ): MeetingLinkResponse

    fun delete(meetingId: String)

    companion object {
        const val OAUTH_CONTENT_TYPE: String = "application/json"
        const val BEARER_TOKEN_TYPE: String = "Bearer"
    }
}
